packages/application/src/plugins/rise.ts

```diff
@@ -0,0 +1,1222 @@
+    url.searchParams.delete('activeCellIndex');
+    url.searchParams.delete('fullscreen');
+    window.history.pushState(null, '', url.toString());
+
+    Promise.all([
+      // Load settings of the JupyterLab extension - so the settings can be edited in JLab.
+      settingRegistry?.load('rise-jupyterlab:plugin') ?? Promise.resolve(null),
+      app.started,
+      app.restored
+    ]).then(async ([settings]) => {
+      let rendered: boolean | null = null;
+      const notebookPath = PageConfig.getOption('notebookPath');
+      const notebookPanel = documentManager.open(notebookPath) as NotebookPanel;
+
+      Rise.registerCommands(app.commands, trans);
+      if (palette) {
+        [
+          CommandIDs.riseFullScreen,
+          CommandIDs.riseHelp,
+          CommandIDs.riseNotesOpen,
+          CommandIDs.riseFirstSlide,
+          CommandIDs.riseLastSlide,
+          CommandIDs.riseShowConfig,
+          CommandIDs.riseChalkboardToggle,
+          CommandIDs.riseChalkboardClear,
+          CommandIDs.riseChalkboardReset,
+          CommandIDs.riseChalkboardColorNext,
+          CommandIDs.riseChalkboardColorPrev
+        ].forEach(command => {
+          palette.addItem({
+            command,
+            category: 'Rise'
+          });
+        });
+      }
+
+      const initializeReveal = (
+        _: any,
+        change: IChangedArgs<any, any, string>
+      ) => {
+        if (
+          change.name === 'dirty' &&
+          change.newValue === false &&
+          // if rendered = null || true
+          !!rendered
+        ) {
+          console.log(`Convert notebook ${notebookPath} to slideshow.`);
+          notebookPanel.content.fullyRendered.disconnect(setRendered, this);
+          notebookPanel.model?.stateChanged.disconnect(initializeReveal, this);
+
+          // Set the active cell index
+          notebookPanel.content.activeCellIndex = activeCellIndex;
+
+          // We wait for the notebook to be loaded to get the settings from the metadata.
+          Rise.loadConfiguration(settings, notebookPanel.model);
+          Rise.revealMode(notebookPanel, app.commands, trans).catch(reason => {
+            console.error(
+              'Fail to update the notebook with Reveal.JS.',
+              reason
+            );
+          });
+
+          Signal.disconnectAll(this);
+        }
+      };
+
+      const setRendered = (notebook: Notebook, fullyRendered: boolean) => {
+        rendered = fullyRendered;
+        if (rendered) {
+          initializeReveal(null, {
+            name: 'dirty',
+            newValue: notebook.model?.dirty ?? true,
+            oldValue: true
+          });
+        }
+      };
+
+      // Deal with virtual rendering
+      notebookPanel.content.fullyRendered.connect(setRendered, this);
+
+      notebookPanel.model?.stateChanged.connect(initializeReveal, this);
+
+      // Remove the toolbar
+      notebookPanel.toolbar.dispose();
+
+      app.shell.add(notebookPanel);
```

Comment:
Maybe

```suggestion
      app.shell.add(notebookPanel, 'rise');
```

see my comment on `packages/application/src/app/index.ts`.